packages/firestore/src/api/user_data_converter.ts

```diff
@@ -478,10 +483,16 @@ export class UserDataConverter {
   /**
    * Parse a "query value" (e.g. value in a where filter or a value in a cursor
    * bound).
+   *
+   * @param allowArrays Whether the dataContext should allow nested arrays.
```

Comment:
dataContext is an internal implementation detail of this method and so it shouldn't be referenced in the method comments (i.e. this is a ["leaky abstraction"](https://en.wikipedia.org/wiki/Leaky_abstraction)).  I'd change it to something like "Whether the query value is an array that may directly contain additional arrays (e.g. the operand of an 'in' query)."

Reply:
oh woah, I did not know there was an official name for that. Thanks for sharing!